modules/apps/27-interchain-accounts/keeper/handshake.go

```diff
@@ -70,7 +86,25 @@ func (k Keeper) OnChanOpenTry(
 	counterpartyVersion string,
 ) error {
 	if order != channeltypes.ORDERED {
-		return sdkerrors.Wrapf(channeltypes.ErrInvalidChannelOrdering, "invalid channel ordering: %s, expected %s", order.String(), channeltypes.ORDERED.String())
+		return sdkerrors.Wrapf(channeltypes.ErrInvalidChannelOrdering, "expected %s channel, got %s", channeltypes.ORDERED, order)
+	}
+
+	if portID != types.PortID {
+		return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "expected %s, got %s", types.PortID, portID)
+	}
+
+	connSequence, err := types.ParseHostConnSequence(counterparty.PortId)
+	if err != nil {
+		return sdkerrors.Wrapf(err, "expected format <app-version%[1]sconn-seq%[1]scounterparty-conn-seq%[1]sowner>, got %s", types.Delimiter, portID)
```

Comment:
Format string here may look a little odd using `%[1]s`, [see explicit argument indexing here](https://pkg.go.dev/fmt#hdr-Explicit_argument_indexes). Avoids having to specify `types.Delimiter` multiple times in args.

Suggestions for a better error message are welcome, but I think including the expected format as @crodriguezvega  mentioned would be valuable.

Reply:
I'd define the expected string as a const and then link the spec in the comment for the expected format. I'd also recommend using `controller-conn-seq` and `host-conn-seq` since counterparty means different things depending on context